hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/ssl/TestReloadingX509TrustManager.java

```diff
@@ -51,16 +52,15 @@ public void testReload() throws Exception {
     TrustManager tm =
         caClient.getServerKeyStoresFactory().getTrustManagers()[0];
     X509Certificate cert1 = caClient.getRootCACertificate();
-    assertEquals(cert1,
-        ((ReloadingX509TrustManager)tm).getAcceptedIssuers()[0]);
+    assertTrue(isCertAcceptedInTrustManager(cert1, tm));
 
     caClient.renewRootCA();
     caClient.renewKey();
     X509Certificate cert2 = caClient.getRootCACertificate();
     assertNotEquals(cert1, cert2);
-
-    assertEquals(cert2,
-        ((ReloadingX509TrustManager)tm).getAcceptedIssuers()[0]);
+    
+    assertTrue(isCertAcceptedInTrustManager(cert1, tm));
```

Comment:
You can use MatcherAsserts here in the form of:
org.hamcrest.MatcherAssert.assertThat(tm.getAcceptedIssuers(), org.hamcrest.Matchers.arrayContaining(cert1)

It is expressive, and you do not have to write your own assertion logic for array contains.